src/framework/theme/components/autocomplete/autocomplete.directive.ts

```diff
@@ -0,0 +1,264 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  forwardRef,
+  Input,
+  OnDestroy,
+  QueryList,
+} from '@angular/core';
+import { NbAutocompleteComponent } from './autocomplete.component';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { NbScrollStrategy } from '../cdk/overlay/mapping';
+import { NbTrigger, NbTriggerStrategy, NbTriggerStrategyBuilderService } from '../cdk/overlay/overlay-trigger';
+import { NbOverlayService } from '../cdk/overlay/overlay-service';
+import { filter, startWith, switchMap, takeWhile, tap } from 'rxjs/operators';
+import { NbOptionComponent } from '../select/option.component';
+import { merge } from 'rxjs';
+import { DOWN_ARROW, ENTER, ESCAPE, UP_ARROW } from '@angular/cdk/keycodes';
+
+@Directive({
+  selector: 'input[nbAutocomplete]',
+  host: {
+    '(input)': 'handleInput($event)',
+    '(keydown)': 'handleKeydown($event)',
+    '(focus)': 'handleFocus($event)',
+    '(blur)': 'handleBlur($event)',
+  },
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => NbAutocompleteDirective),
+    multi: true,
+  }],
+
+})
+export class NbAutocompleteDirective<T> implements AfterViewInit, OnDestroy, ControlValueAccessor {
+
+  protected autocomplete: NbAutocompleteComponent<T>;
+
+  protected isOpenedAfterFocus: boolean = false;
+
+  /**
+   * Trigger strategy used by overlay.
+   * */
+  protected triggerStrategy: NbTriggerStrategy;
+
+  /**
+   * Provides autocomplete component.
+   * */
+  @Input('nbAutocomplete')
+  set setAutocomplete(autocomplete: NbAutocompleteComponent<T>) {
+    this.autocomplete = autocomplete;
+    this.setupAutocomplete();
+  }
+
+  constructor(
+    protected hostRef: ElementRef,
+    protected overlay: NbOverlayService,
+    protected cd: ChangeDetectorRef,
+    protected triggerStrategyBuilder: NbTriggerStrategyBuilderService) {}
+
+  ngAfterViewInit() {
+    this.triggerStrategy = this.createTriggerStrategy();
+    this.subscribeOnTriggers();
+  }
+
+  protected handleInput($event: any) {
+    const currentValue = this.hostRef.nativeElement.value;
+    this._onChange(currentValue);
+    this.hostRef.nativeElement.value = this.getDisplayValue(currentValue);
```

Comment:
There is method `setHostInputValue` for setting host element value

Reply:
Fixed